scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -3409,9 +3451,76 @@ def __get_log_config_for_container(self, cid, info, k8s_cache, base_attributes):
         if "parser" in attrs:
             result["parser"] = attrs["parser"]
 
-        return result
+        results = [result]
+
+        # Based on the pod annotations in a format {container_name}.{team}.secret={secret_name}
+        # we might want to add api_keys parameter
+        if container_annotations or all_annotations:
+            api_keys = self.__container_api_keys_from_annotations(k8s_cache, pod_namespace, pod_name, container_annotations, all_annotations)
+
+            if api_keys:
+                results = [
+                    {**result, "api_key": api_key}
+                    for api_key in list(filter(lambda api_key: api_key is not None, api_keys))
+                ]
+
+        return results
+
+    def __container_api_keys_from_annotations(self, k8s_cache, pod_namespace, pod_name, container_annotations, all_annotations):
+        def fetch_secret(name):
+            secret = k8s_cache.secret(self.__agent_pod.namespace, name, time.time())
+            if secret:
+                return secret
+
+            self._logger.warning(
+                "Failed to fetch secret '%s' for pod '%s/%s', ignoring."
+                % (name, pod_namespace, pod_name),
+                limit_once_per_x_secs=300,
+                limit_key="k8s-fetch-secret-%s" % name,
+            )
+
+        def get_secret_api_key(secret):
+            api_key = base64.b64decode(secret.data.get("scalyr-api-key")).decode("utf-8")
+
+            if not api_key:
+                self._logger.warning(
+                    "Secret '%s/%s' does not contain a scalyr-api-key field, ingoring."
+                    % (pod_namespace, secret.name),
+                    limit_once_per_x_secs=300,
+                    limit_key="k8s-fetch-secret-%s" % secret.name
+                )
+
+            return api_key
+
+        container_teams = container_annotations.get("teams", {})
+        default_teams = all_annotations.get("teams", {})
+
+        secret_names = [
+            team["secret"]
+            for team in container_teams
+            if "secret" in team
+        ]
+
+        if not secret_names:
+            secret_names = [
+                team["secret"]
+                for team in default_teams
+                if "secret" in team
+            ]
+
+        api_keys = [
+            get_secret_api_key(
+                fetch_secret(
+                    secret_name
+                )
+            )
+            for secret_name in secret_names
+        ]
```

Comment:
I think it's better to filter out api_key = None here rather than in line 3464

Reply:
Definitely!